webapp/src/components/DynamicButtonDataTable.vue

```diff
@@ -0,0 +1,201 @@
+<template>
+  <div class="button-group d-flex justify-content-between align-items-center">
+    <div class="button-left">
+      <button
+        v-if="dataType === 'samples'"
+        class="btn btn-default"
+        @click="$emit('open-create-item-modal')"
+      >
+        Add an item
+      </button>
+      <button
+        v-if="dataType === 'samples'"
+        class="btn btn-default ml-2"
+        @click="$emit('open-batch-create-item-modal')"
+      >
+        Add batch of items
+      </button>
+      <button
+        v-if="dataType === 'samples'"
+        class="btn btn-default ml-2"
+        @click="$emit('open-qr-scanner-modal')"
+      >
+        <font-awesome-icon icon="qrcode" /> Scan QR
+      </button>
+      <button
+        v-if="dataType === 'collections'"
+        class="btn btn-default"
+        @click="$emit('open-create-collection-modal')"
+      >
+        Create new collection
+      </button>
+      <button
+        v-if="dataType === 'startingMaterials' && editableInventory"
+        class="btn btn-default"
+        :disabled="itemsSelected.length === 0"
+        @click="$emit('open-create-item-modal')"
+      >
+        Add a starting material
+      </button>
+      <button
+        v-if="dataType === 'equipments'"
```

Comment:
```suggestion
        v-if="dataType === 'equipment'"
```

You get the idea, I think all of the `equipments` should be changed to `equipment` throughout.